File: week12/ClockAppTemplate/ClockAppTemplate/Program.cs
using System;
using System.Diagnostics;

namespace ClockApp
{
    public class Program
    {
        private static void Main(string[] args)
        {
            Stopwatch stopwatch = Stopwatch.StartNew();
            Clock MyClock = new Clock();
            for (int i = 0; i < 10000; i++)
            {
                MyClock.Tick();
            }

            //Get the current process
            System.Diagnostics.Process proc = System.Diagnostics.Process.GetCurrentProcess();
            Console.WriteLine("Current process: {0}", proc.ToString());
            //Display the total physical memory size allocated for the current process
            Console.WriteLine("Physical memory usage: {0} bytes", proc.WorkingSet64);
            // Display peak memory statistics for the process.
            Console.WriteLine("Peak physical memory usage {0} bytes", proc.PeakWorkingSet64);
            //extract memory

            //extract the timelapse
            Console.WriteLine(MyClock.GetTime());
            MyClock.Restart();
            Console.WriteLine(MyClock.GetTime());
            stopwatch.Stop();
        }
    }
}

File: week12/ClockAppTemplate/ClockAppTemplate/program.py
import tracemalloc
import time
from clock import Clock

def main():
    my_clock = Clock()
    num_ticks = 10000 

    tracemalloc.start() # Starts monitoring memory usage

    for _ in range(num_ticks):
        my_clock.tick()

    print(f"Current and peak memory used: {tracemalloc.get_traced_memory()}")

    tracemalloc.stop() 

    print(f"Time: {my_clock.get_time()}")
    
    my_clock.reset()
    print(f"Time after reset: {my_clock.get_time()}")



main()